test/e2e/gardener/shoot/internal/node/node.go

```diff
@@ -0,0 +1,156 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"context"
+	"time"
+
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	"github.com/gardener/gardener/test/framework"
+	"github.com/gardener/gardener/test/utils/shoots/access"
+)
+
+const name = "e2e-test-node-critical"
+
+// VerifyNodeCriticalComponentsBootstrapping tests the node readiness feature (see docs/usage/node-readiness.md).
+func VerifyNodeCriticalComponentsBootstrapping(ctx context.Context, f *framework.ShootFramework) {
+	shootClientSet, err := access.CreateShootClientFromAdminKubeconfig(ctx, f.GardenClient, f.Shoot)
+	Expect(err).To(Succeed())
+
+	var (
+		seedClient  = f.SeedClient.Client()
+		shootClient = shootClientSet.Client()
+		technicalID = f.Shoot.Status.TechnicalID
+	)
+
+	By("Create ManagedResource for shoot with broken node-critical component")
+	createOrUpdateNodeCriticalManagedResource(ctx, seedClient, shootClient, technicalID, "non-existing")
+	DeferCleanup(func() {
+		cleanupCtx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+
+		cleanupNodeCriticalManagedResource(cleanupCtx, seedClient, technicalID)
+	})
+
+	By("Delete Nodes and Machines to trigger new Node bootstrap")
+	Eventually(func(g Gomega) {
+		g.Expect(seedClient.DeleteAllOf(ctx, &machinev1alpha1.Machine{}, client.InNamespace(technicalID))).To(Succeed())
+		g.Expect(shootClient.DeleteAllOf(ctx, &corev1.Node{})).To(Succeed())
+	}).Should(Succeed())
+
+	By("Wait for new Node to be created")
+	var node *corev1.Node
+	Eventually(func(g Gomega) {
+		nodeList := &corev1.NodeList{}
+		g.Expect(shootClient.List(ctx, nodeList)).To(Succeed())
+		g.Expect(nodeList.Items).NotTo(BeEmpty(), "new Node should be created")
+		node = &nodeList.Items[0]
+	}).WithContext(ctx).WithTimeout(2 * time.Minute).Should(Succeed())
```

Comment:
Is this timeout large enough? Maybe we should better go with `5m` or even `10m`?